altium.py

```diff
@@ -831,22 +835,76 @@ def handle_unknown(self, objects, obj):
             obj.get(property)
         obj.check("DESIMPCOUNT", b"1", None)
     
-    @_setitem(handlers, 16)
-    def handle_unknown(self, objects, obj):
-        for [name, value] in (
-            ("AREACOLOR", b"8454143"), ("ARROWKIND", b"Block & Triangle"),
-            ("COLOR", b"128"), ("OWNERPARTID", b"-1"),
-            ("TEXTCOLOR", b"128"), ("TEXTFONTID", b"1"),
-            ("TEXTSTYLE", b"Full"),
+    @_setitem(handlers, Record.SHEET_ENTRY) # id=16
+    def handle_sheetport(self, objects, obj):
+        for property in (
+            "OWNERPARTID", "INDEXINSHEET", "ARROWKIND", "STYLE", "TEXTSTYLE"
         ):
-            obj.check(name, value)
-        obj.get("HARNESSTYPE")
-        obj.check("SIDE", None, b"1")
-        obj.get_int("INDEXINSHEET")
-        obj.get_int("DISTANCEFROMTOP")
-        obj.get_int("STYLE")
-        obj.get_int("IOTYPE")
-        obj.get("NAME")
+            obj.get(property)
+            
+        if self.last_sheet_symbol:
+            with self.renderer.view(offset=get_location(self.last_sheet_symbol)) as view:
```

Comment:
I wonder if you can get automatic access to the parent sheet symbol as objects.properties, i.e. you could call `get_location(objects.properties)` etc. I think “objects” was meant to be the parent or owner object. I could probably use a better name :)